app/controllers/email_confirmations_controller.rb

```diff
@@ -17,10 +19,37 @@ def mfa_update
     if @user.mfa_enabled? && @user.otp_verified?(params[:otp])
       confirm_email
     else
-      @form_url       = mfa_update_email_confirmations_url(token: @user.confirmation_token)
-      flash.now.alert = t("multifactor_auths.incorrect_otp")
-      render template: "multifactor_auths/otp_prompt", status: :unauthorized
+      @form_mfa_url = mfa_update_email_confirmations_url(token: @user.confirmation_token)
+      login_failure(t("multifactor_auths.incorrect_otp"))
+    end
+  end
+
+  def webauthn_update
+    @user = User.find(session.dig(:webauthn_authentication, "user"))
+    @challenge = session.dig(:webauthn_authentication, "challenge")
+
+    if params[:credentials].blank?
+      login_failure("Credentials required")
```

Comment:
Yeah this "Credentials required" is seen in a few controllers without translation. I'll make a ticket to add a translation for it. I don't think I want to do that in this PR because it will make it even more chonky.